.devcontainer/features/languages/nodejs/install.sh

```
@@ -0,0 +1,316 @@
+  warn() { echo -e "${YELLOW}⚠${NC} $*"; }
+}
+
+# Retry function
+retry() {
+  local max_attempts=$1
+  local delay=$2
+  shift 2
+  local attempt=1
+  local exit_code=0
+
+  while [ $attempt -le $max_attempts ]; do
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)